CrazyCanvas/Source/GUI/HUDGUI.cpp

```diff
@@ -192,6 +195,183 @@ void HUDGUI::DisplayHitIndicator()
+
+	// Name is different than other labels
+	Ptr<Label> pNameLabel = *new Label();
+
+	Ptr<SolidColorBrush> pWhiteBrush = *new SolidColorBrush();
+	pWhiteBrush->SetColor(Color::White());
+
+	pNameLabel->SetContent(newPlayer.GetName().c_str());
+	pNameLabel->SetForeground(pWhiteBrush);
+	pNameLabel->SetFontSize(28.f);
+	pNameLabel->SetVerticalAlignment(VerticalAlignment::VerticalAlignment_Bottom);
+	uint8 column = 1;
+	pGrid->GetChildren()->Add(pNameLabel);
+	pGrid->SetColumn(pNameLabel, column++);
+
+	AddStatsLabel(pGrid, std::to_string(newPlayer.GetKills()), column++);
+	AddStatsLabel(pGrid, std::to_string(newPlayer.GetDeaths()), column++);
+	AddStatsLabel(pGrid, std::to_string(newPlayer.GetFlagsCaptured()), column++);
+	AddStatsLabel(pGrid, std::to_string(newPlayer.GetFlagsDefended()), column++);
+	AddStatsLabel(pGrid, std::to_string(newPlayer.GetPing()), column++);
+
+	if (newPlayer.GetTeam() == 0)
+	{
+		m_pBlueTeamStackPanel->GetChildren()->Add(pGrid);
+	}
+	else if (newPlayer.GetTeam() == 1)
+	{
+		m_pRedTeamStackPanel->GetChildren()->Add(pGrid);
+	}
+	else
+	{
+		LOG_WARNING("[HUDGUI]: Unknown team on player \"%s\".\n\tUID: %lu\n\tTeam: %d",
+			newPlayer.GetName().c_str(), newPlayer.GetUID(), newPlayer.GetTeam());
+	}
+
+	if (PlayerManagerClient::GetPlayerLocal()->GetUID() == newPlayer.GetUID())
+	{
+		Ptr<Image> localPlayerIcon = *new Image();
+		Ptr<BitmapImage> srcImage = *new BitmapImage();
+		srcImage->SetUriSource(Noesis::Uri::Uri("splashes/splash_green.png"));
+		localPlayerIcon->SetSource(srcImage);
+		localPlayerIcon->SetWidth(40);
+		localPlayerIcon->SetHeight(40);
+		pGrid->GetChildren()->Add(localPlayerIcon);
+		pGrid->SetColumn(localPlayerIcon, 0);
+	}
+}
+
+void HUDGUI::RemovePlayer(const Player& player)
+{
+	Grid* pGrid = FrameworkElement::FindName<Grid>(std::to_string(player.GetUID()).c_str());
+
+	if (!pGrid)
+	{
+		LOG_WARNING("[HUDGUI]: Tried to delete \"%s\", but could not find grid.\n\tUID: %lu",
+			player.GetName().c_str(), player.GetUID());
+		return;
+	}
+
+	if (player.GetTeam() == 0)
+	{
+		m_pBlueTeamStackPanel->GetChildren()->Remove(pGrid);
+	}
+	else if (player.GetTeam() == 1)
+	{
+		m_pRedTeamStackPanel->GetChildren()->Remove(pGrid);
+	}
+}
+
+void HUDGUI::UpdatePlayerProperty(uint64 playerUID, EPlayerProperty property, const LambdaEngine::String& value)
+{
+	uint8 index = 0;
+	switch (property)
+	{
+	case EPlayerProperty::PLAYER_PROPERTY_NAME:				index = 0; break;
+	case EPlayerProperty::PLAYER_PROPERTY_KILLS:			index = 1; break;
+	case EPlayerProperty::PLAYER_PROPERTY_DEATHS:			index = 2; break;
+	case EPlayerProperty::PLAYER_PROPERTY_FLAGS_CAPTURED:	index = 3; break;
+	case EPlayerProperty::PLAYER_PROPERTY_FLAGS_DEFENDED:	index = 4; break;
+	case EPlayerProperty::PLAYER_PROPERTY_PING:				index = 5; break;
+	default: LOG_WARNING("[HUDGUI]: Enum not supported"); return;
+	}
+
+	Grid* pGrid = FrameworkElement::FindName<Grid>(std::to_string(playerUID).c_str());
```

Comment:
I think it could be good to save these upon creation, that way you won't have to search for this string every time this function is invoked.

Reply:
Yes fixed 🤠